### PR TITLE
Assigned user to task

### PR DESCRIPTION
**Pull request by ShevchukTania**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [ ] code coverage >90%
- [x] DB indexes added if needed
- [x] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

image

image

image

### CODE REVIEW

File: app/controllers/account/tasks_controller.rb [hunk 1]
```diff
@@ -60,8 +60,28 @@ def watch
...
+    @task = @project.tasks.find(params[:id])
+    respond_to :js
+  end
+
+  def assign
+    @project = parent
+    @task = @project.tasks.find(params[:id])
+    @assignee = @task.build_assignee(assignee_params)
+    if @assignee.save
+      respond_to :js
```
Comment (romaslmd) [inline app/controllers/account/tasks_controller.rb]: not a very good idea
js response should be for both valid and invalid requests
for invalid should be at least a flash message

File: app/controllers/account/tasks_controller.rb [hunk 2]
```diff
@@ -104,4 +124,8 @@ def task_movement_params
   def section_params
     params.permit(:section)
   end
+
+  def assignee_params
+    params.require(:assignee).permit(:user_id).merge!(task_id: params[:id])
```
Comment (romaslmd) [inline app/controllers/account/tasks_controller.rb]: u're using a built-in method to build an assoc, thus task_id should already be assigned

File: app/models/task.rb [hunk 3]
```diff
@@ -31,6 +7,8 @@ class Task < ApplicationRecord
   belongs_to :project, required: true
   has_many :task_watches, dependent: :destroy
   has_many :watchers, through: :task_watches, source: :user
+  has_one :assignee, dependent: :destroy
+  has_one :assigned_user, through: :assignee, source: :user
```
Comment (romaslmd) [inline app/models/task.rb:38]: nice :+1:

File: spec/controllers/account/tasks_controller_spec.rb [hunk 4]
```diff
@@ -143,4 +144,41 @@
     end
   end
 
+  context '#GET / choose_assignee' do
```
Comment (romaslmd) [inline spec/controllers/account/tasks_controller_spec.rb]: remove spaces after /

File: config/initializers/simple_form.rb [hunk 5]
```diff
@@ -1,12 +1,3 @@
-# frozen_string_literal: true
```
Comment (romaslmd) [inline config/initializers/simple_form.rb]: you'd better leave this line

File: app/assets/stylesheets/main.scss [hunk 6]
```diff
@@ -15,4 +15,24 @@
 .footer-content {
   padding: 10px 10px;
 }
+div#assingnee_user_bar{
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/main.scss]: create maybe tasks.scss file and store these styles there

File: app/controllers/account/tasks_controller.rb (hunk 1 above)
Comment (igorkasyanchuk) [reply app/controllers/account/tasks_controller.rb]: yes, respond even if @assignee is not saved

File: app/models/assignee.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+class Assignee < ApplicationRecord
```
Comment (igorkasyanchuk) [inline app/models/assignee.rb]: why so complex ...

rework it. remove this association.

Task has only one assignee. So task `belongs_to :assignee, class_name: "User"`

File: app/models/task.rb [hunk 8]
```diff
@@ -31,6 +7,8 @@ class Task < ApplicationRecord
...
```
Comment (igorkasyanchuk) [inline app/models/task.rb]: rework relations

File: app/views/account/tasks/_assignee_show.html.slim [hunk 9]
```diff
@@ -0,0 +1,15 @@
+  #assignee_show
+    = fa_icon 'user-o'
+    -if task.assigned_user
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/_assignee_show.html.slim]: add space before "if"

File: app/views/account/tasks/_assignee_show.html.slim [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+      span Assigned to:
+      =  link_to choose_assignee_account_project_task_path(task.project),
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/_assignee_show.html.slim]: remove extra space

File: app/views/account/tasks/_assignee_show.html.slim [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+                remote: true, id: 'assignee_show' do
+        = task.assigned_user.full_name
+
+      = link_to unassign_account_project_task_path(task.project),
+                method: :delete, remote: true, :data => {:confirm => 'Are you sure?'} do
+                  = fa_icon 'times'
+    -else
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/_assignee_show.html.slim]: add space

File: app/views/account/tasks/assign.js.erb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+$('.notification').remove();
+<% if @result %>
+  $('#assignee_form').html('<%=j render "account/tasks/assignee_show", task: @task %>')
+<% else %>
+  <% flash[:alert] = @task.errors.full_messages.to_sentence %>
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/assign.js.erb:5]: why here you don't have code like you have below

$('body').prepend("<%= j render 'shared/flash' %>");

add here or remove in other place

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**